All project code/index.js

```diff
@@ -79,32 +86,58 @@ app.get("/login", (req, res) => {
 });
 
 // Handles user login from a /login POST request.
-app.post("/login", (req, res) => {
+app.post("/login", async (req, res) => {
     const username = req.body.username;
     const password = req.body.password;
-
     const query = "SELECT * FROM users WHERE username = $1";
 
-    db.any(query, [username])
-    .then(async (user) => {
-        const match = await bcrypt.compare(password, user[0].password);
-
-        if (match)
-        {
-            req.session.user = "TEST";
-            req.session.save();
-            res.redirect("/main");
-        }
-        else
-        {
-            console.log("Incorrect username or password.");
-            res.redirect("/login");
-        }
+    let user; // The user found (or not found) in the database
+
+    // Queries database to find matching user for the inputted username.
+    await db.any(query, [username])
+    .then((foundUser) => {
+        user = foundUser[0];
     })
     .catch((err) => {
-        console.log(err.message);
+        // Something went wrong, refresh the page for the user to try again.
         res.redirect("/login");
+        return console.log(err.message);
```

Comment:
This would work, but isn't really standard per-say. It's not really normal to return a console.log message. I'm not even sure this line would execute because the res.redirect is essentially a function call that doesn't return so it shouldn't even reach that line.

Reply:
The early return should be there for when res.redirect finishes. Keep in mind, these are asynchronous calls, and this early return stops some unintended race conditions.

As for returning console.log, I saw this being done elsewhere, and it likely doesn't matter. I can remove that